skywalking/plugins/sw_urllib_request.py

```diff
@@ -38,15 +39,22 @@ def _sw_open(this: OpenerDirector, fullurl, data, timeout):
 
         context = get_context()
         carrier = Carrier()
-        with context.new_exit_span(op=fullurl.selector or '/', peer=fullurl.host, carrier=carrier) as span:
+        url = fullurl.selector.split("?")[0] if fullurl.selector else '/'
+        with context.new_exit_span(op=url, peer=fullurl.host, carrier=carrier) as span:
             span.layer = Layer.Http
             span.component = Component.General
 
             [fullurl.add_header(item.key, item.val) for item in carrier]
 
-            res = _open(this, fullurl, data, timeout)
             span.tag(Tag(key=tags.HttpMethod, val=fullurl.get_method()))
             span.tag(Tag(key=tags.HttpUrl, val=fullurl.full_url))
+
+            try:
+                res = _open(this, fullurl, data, timeout)
```

Comment:
I think this breaks the tests, `fullurl.get_method()` is not static, it differs before/after `_open(this, fullurl, data, timeout)`